src/test/MockConsentClient.ts

```diff
@@ -0,0 +1,27 @@
+import type { ConsentClient } from "@/types";
+
+/**
+ * A mock implementation of the ConsentClient interface
+ *
+ * This class can be used in place of OakConsentClient in tests
+ */
+export class MockConsentClient implements ConsentClient {
+  appSlug = "testApp";
+  userId = "testUserId";
+  init() {
+    return Promise.resolve();
+  }
+  getState() {
+    return {
+      policyConsents: [],
+      requiresInteraction: true,
+    };
+  }
+  onStateChange = () => {
+    return () => {};
+  };
+  logConsents = () => {
+    return Promise.resolve();
+  };
+  getConsent = () => "pending" as const;
+}
```

Comment:
This can be used to simplify the implementation in the host app (particularly OWA) where we have a wrapper provider that only exists to make testing easier.